teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java

```diff
@@ -173,4 +169,50 @@ private Path convertToPath(final String option) {
     }
     return Path.of(option);
   }
+
+  static class PublicKeyLoader {
+    final ObjectMapper objectMapper;
+
+    public PublicKeyLoader() {
+      this(new ObjectMapper());
+    }
+
+    public PublicKeyLoader(final ObjectMapper objectMapper) {
+      this.objectMapper = objectMapper;
+    }
+
+    public List<BLSPublicKey> getPublicKeys(final List<String> publicKeys) {
+      if (publicKeys == null || publicKeys.isEmpty()) {
+        return Collections.emptyList();
+      }
+
+      try {
+        final Set<BLSPublicKey> blsPublicKeySet =
+            publicKeys.stream()
+                .filter(key -> !key.contains(":"))
+                .map(key -> BLSPublicKey.fromSSZBytes(Bytes.fromHexString(key)))
+                .collect(Collectors.toSet());
+        blsPublicKeySet.addAll(
+            publicKeys.stream()
+                .filter(key -> key.contains(":"))
+                .flatMap(this::readKeysFromUrl)
+                .collect(Collectors.toList()));
+
+        return List.copyOf(blsPublicKeySet);
+      } catch (IllegalArgumentException e) {
+        throw new InvalidConfigurationException(
+            "Invalid configuration. Signer public key is invalid", e);
+      }
+    }
+
+    private Stream<BLSPublicKey> readKeysFromUrl(final String url) {
+      try {
+        final String[] keys = objectMapper.readValue(new URL(url), String[].class);
+        return Arrays.asList(keys).stream()
+            .map(key -> BLSPublicKey.fromSSZBytes(Bytes.fromHexString(key)));
+      } catch (IOException ex) {
+        throw new IllegalArgumentException("Failed to load public keys from URL", ex);
```

Comment:
Need to test that this provide a friendly error message - possibly needs to throw `InvalidConfigurationException` to make that happen.